mcp/src/minecraft/com/elementfx/tvp/ad/block/BlockCustomBed.java

```diff
@@ -64,4 +79,77 @@ public ItemStack getItem(World worldIn, BlockPos pos, IBlockState state)
     {
         return new ItemStack(this.inventoryItem);
     }
+
+    public void addCollisionBoxToList(IBlockState state, World worldIn, BlockPos pos, AxisAlignedBB entityBox, List<AxisAlignedBB> collidingBoxes, @Nullable Entity entityIn)
+    {
+        state = state.getActualState(worldIn, pos);
+        addCollisionBoxToList(pos, entityBox, collidingBoxes, BED_AABB);
+
+        if (((Boolean)state.getValue(BUNK)).booleanValue())
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, BUNK_AABB);
+        }
+
+    }
+    
+    public AxisAlignedBB getBoundingBox(IBlockState state, IBlockAccess source, BlockPos pos)
+    {
+        state = this.getActualState(state, source, pos);
+        return BOUNDING_BOXES[getBoundingBoxIdx(state)];
+    }
+
+    /**
+     * Returns the correct index into boundingBoxes, based on what the fence is connected to.
+     */
+    private static int getBoundingBoxIdx(IBlockState state)
+    {
+        int i = 0;
+
+        if (((Boolean)state.getValue(BUNK)).booleanValue())
+        {
+            i |= 1;
+        }
+
+        return i;
+    }
+
+    public boolean canConnectTo(IBlockAccess worldIn, BlockPos pos)
+    {
+    	IBlockState iblockstate = worldIn.getBlockState(pos);
+        Block block = iblockstate.getBlock();
+        IBlockState iblockstateup = worldIn.getBlockState(pos.up());
+        Block blockup = iblockstateup.getBlock();
+        
+        if(!(blockup instanceof BlockCustomBed && block instanceof BlockCustomBed))
+        {
+        	return false;
+        }
+        
+        Boolean isSameDirection = iblockstate.getValue(FACING) == iblockstateup.getValue(FACING);
+        Boolean isSamePart = iblockstate.getValue(PART) == iblockstateup.getValue(PART);
+
+        return isSamePart.booleanValue() == isSameDirection.booleanValue();
```

Comment:
Careful here. I suggested the following situations for connecting:
1. The block above is a BlockCustomBed with the `state.FACING == stateOther.FACING` and `state.PART == stateOther.PART`
2. The block above is a BlockCustomBed with the `state.FACING == stateOther.FACING.getOpposite()` and `state.part != stateOther.PART`

What you have would be the same as above if `state.FACING == stateOther.FACING.getOpposite()` and `state.FACING != stateOther.FACING` were the same. However if a block is facing perpendicular to the current direction, the facing is not equal to the other or its opposite.